rfcs/convergence/props-consistency-check.md

```diff
@@ -0,0 +1,535 @@
+# Consistency Check
+
+The following components and their props were reviewed as of 9/7/2021 for consistency.
+
+Accordian, AccordianHeader, AccordianItem, AccordianPanel
+Avatar
+Badge, CounterBadge, PresenceBadge,
+Button, CompoundButton, MenuButton, ToggleButton,
+Divider,
+Image,
+Label,
+Link,Menu, MenuDivider, MenuGroup, MenuGroupHeader,MenuItem, MenuItemCheckbox, MenuItemRadio, MenuList, MenuPopover, MenuTrigger
+Popover, PopoverSurface, PopoverTrigger
+Portal
+
+## Nit: Explicit component type declaration
+
+> Some components declare their type explicity, others do not.
+
+**Recommendations**
+
+- Based on [forwardRef PR](https://github.com/microsoft/fluentui/pull/19812) from Ben, prefer allowing the component types to be inferred
+
+**Proposed Work Items**
+
+- [ ] Update components to use new forwardRef
+
+**Details**
+
+```tsx
+export const Accordion: React.FunctionComponent<AccordionProps & React.RefAttributes<HTMLElement>> = React.forwardRef<
+  HTMLElement,
+  AccordionProps
+>((props, ref) =>
+```
+
+```tsx
+export const Avatar = React.forwardRef((props: AvatarProps, ref: React.Ref<HTMLElement>) =>
+```
+
+## Props declared as interfaces vs types
+
+> Some props are defined as interfaces and extended. Others are defined as types and intersected. This isn't always consistent within the same component.
+
+**Documentation**
+
+- [TS docs](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#differences-between-type-aliases-and-interfaces)
+- [TS playgroud](https://www.typescriptlang.org/play?q=414#example/types-vs-interfaces) recommends interfaces, but may be outdated :
+  - interfaces provide better error messages
+  - you can extend an interface by declaring it a second time
+- [StackOverflow thread](https://stackoverflow.com/questions/37233735/interfaces-vs-types-in-typescript/52682220#52682220) adds more information:
+  - Type aliases can be used for primitives, unions, and tuples.
+  - Classes canot implement a type alias that use sa union type.
+- [Team discussion](https://github.com/microsoft/fluentui/pull/19376#discussion_r700046363)
+  - Types can be inlined leading to inflated declaration files
+  - Non-exported interfaces required by a declaration give an error (because they can't be inlined)
+  - No clear decision at the time.
+
+**Recommendations**
+
+- Use types and intersections over interfaces.
+  - Even though Typescript recommends interfaces, their recommendation is in the context of using classes. React hooks is functional programming and I think types are the expected approach.
+  - Unless the types cause so much duplication that we have a bundle size problem, I think they are worth it especially with the union problem.
```

Comment:
There shouldn't be a bundlesize problem. If there was an app that bundled types into their bundle that would definitely be a first.

Generally bundlers like webpack use loaders that will just automatically strip out types. If you use babel, it straight up ignores types